app/views/my/password.html.erb

```diff
@@ -26,14 +26,35 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 See doc/COPYRIGHT.rdoc for more details.
 
 ++#%>
-
 <% html_title l(:label_my_account), l(:button_change_password) %>
 <%= toolbar title: l(:button_change_password) %>
 
 <%= error_messages_for 'user' %>
 <%= styled_form_tag({:action => :change_password},
-                    {:autocomplete => 'off'}) do %>
+                    {:autocomplete => 'off', class: 'change_password'}) do %>
   <%= back_url_hidden_field_tag %>
+<script type="text/javascript">
+  jQuery(document).ready(function(){
+    new showTooltips({
+      element            : '.form--field-container input[name^="new"]',
+    });
+
+    jQuery('#new_password_confirmation, #new_password').keyup(function() {
+      var password_present = (jQuery('#password').val().length > 0);
+      var button = jQuery('input[name="commit"]');
+      var new_pass = jQuery('#new_password').val();
+      var new_pass_confirm = jQuery('#new_password_confirmation').val();
+      var new_passwords_matches = (new_pass === new_pass_confirm);
+      var new_passwords_present = (new_pass.length > 0) && (new_pass_confirm.length > 0);
+
+      if  (password_present && new_passwords_matches && new_passwords_present) {
+        button.removeAttr('disabled');
+      } else {
+        button.attr('disabled','disabled');
+      }
+    })
+  });
+</script>
```

Comment:
Noooooooooooooooooooooooooo.

Why do you introduce the initialization in a template? This should be initialized via a class. And even if you do not do that, please extract this into a custom js file (which still would not be ideal, but better than *this*.